tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalJBossBundle.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.server.core.portal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IPath;
+
+/**
+ * @author Simon Jiang
+ */
+public class PortalJBossBundle extends AbstractPortalBundle  implements PortalBundle
+{
+    
+    public static final int DEFAULT_JMX_PORT = 2099;
+    
+    public PortalJBossBundle( IPath path )
+    {
+       super(path);
+    }
+
+    protected int detectJmxRemotePort()
+    {
+        return DEFAULT_JMX_PORT;
+    }
+    
+    public String getClassToLaunch()
+    {
+        return "org.jboss.modules.Main";
+    }
+
+    protected IPath getPortalDir( IPath appServerDir )
+    {
+        IPath retval = null;
+
+        if( appServerDir != null )
+        {
+            retval = appServerDir.append( "/standalone/deployments/ROOT.war" );
+        }
+
+        return retval;
+    }
+
+    public IPath[] getRuntimeClasspath()
+    {
+        final List<IPath> paths = new ArrayList<IPath>();
+
+        if( this.bundlePath.toFile().exists() )
+        {
+            paths.add( bundlePath.append( "jboss-modules.jar" ) );
+
+        }
+
+        return paths.toArray( new IPath[0] );
+    }
+
+    @Override
+    public String[] getRuntimeStartProgArgs()
+    {
+        final List<String> args = new ArrayList<String>();
+        
+        args.add( "-mp " + "\"" + this.bundlePath.toPortableString() +  "/modules" + "\"" );
+        args.add( "-jaxpmodule javax.xml.jaxp-provider" );
```

Comment:
args.add( "-mp");
args.add("\"" + this.bundlePath.toPortableString() +  "/modules" + "\"");
args.add("-jaxpmodule");
args.add("javax.xml.jaxp-provider");